plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -724,6 +731,20 @@ public void setNull(PreparedStatement statement, int index)
         };
     }
 
+    private static ColumnMapping longVarcharColumnMapping(int columnSize)
+    {
+        // Disable pushdown to avoid "The data types ntext and nvarchar are incompatible in the equal to operator." error
```

Comment:
I tried reproducing this and it seems the issue happens when the column is of type NTEXT only. Since NTEXT column cannot be compared against NVARCHAR literal.

So can we check jdbcTypeName here and disable pushdown only if the column's type is NTEXT?

Reply:
How did you test it? When I tested `text` type (not ntext), I received the below message. 
```
The data types text and nvarchar are incompatible in the equal to operator.
```

Reply:
For NVARCHAR column type it's working. I didn't test with TEXT.

Reply:
This `longVarcharColumnMapping()` method isn't used for `NVARCHAR`.

Reply:
Ah, indeed. You are correct. NVARCHAR pushdown is still working on this branch.